pkg/cmd/build/build.go

```diff
@@ -405,60 +410,74 @@ func GetVolumesToInclude(volumesToInclude []model.StackVolume) []model.StackVolu
 	return result
 }
 
-// parseTempSecrets reads the content of the src of a secret and replaces the envs to mount into dockerfile
-func parseTempSecrets(secretTempFolder string, buildOptions *types.BuildOptions) error {
+// replaceSecretsSourceEnvWithTempFile reads the content of the src of a secret and replaces the envs to mount into dockerfile
+func replaceSecretsSourceEnvWithTempFile(fs afero.Fs, secretTempFolder string, buildOptions *types.BuildOptions) error {
 	// for each secret at buildOptions extract the src
 	// read the content of the file
-	// create a new file under tempFolder with the expanded content
+	// create a new file under secretTempFolder with the expanded content
 	// replace the src of the secret with the tempSrc
 	for indx, s := range buildOptions.Secrets {
-		splitSecret := strings.SplitN(s, "src=", 2)
-		if len(splitSecret) <= 0 {
-			return fmt.Errorf("unable to get secret src")
-		}
-		srcFileName := strings.TrimSpace(splitSecret[1])
-
-		// read source file
-		srcFile, err := os.Open(srcFileName)
+		csvReader := csv.NewReader(strings.NewReader(s))
+		fields, err := csvReader.Read()
 		if err != nil {
-			return err
+			return fmt.Errorf("error reading the csv secret, %w", err)
 		}
 
-		// create temp file
-		tmpfile, err := os.CreateTemp(secretTempFolder, "secret-")
-		if err != nil {
-			return err
-		}
-		writer := bufio.NewWriter(tmpfile)
-
-		sc := bufio.NewScanner(srcFile)
-		for sc.Scan() {
-			// expand content
-			srcContent, err := model.ExpandEnv(sc.Text(), true)
-			if err != nil {
-				return err
+		newFields := make([]string, len(fields))
+		for indx, field := range fields {
+			key, value, found := strings.Cut(field, "=")
+			if !found {
+				return fmt.Errorf("secret format error")
 			}
 
-			// save expanded to temp file
-			_, err = writer.Write([]byte(fmt.Sprintf("%s\n", srcContent)))
-			if err != nil {
-				return fmt.Errorf("unable to write to temp file: %s", err)
+			if key == "src" || key == "source" {
```

Comment:
Should this be `&&`?

Reply:
its source or src indicating the source of the secret, i guess is OR since the two wont be able to exist at the same time